### DISCUSSION TITLE
如何获取当前光标位置？并将自定义工具栏返回的数据插入指定光标位置？

### DISCUSSION THREAD

**Discussion by Nabian31** [imzbf/md-editor-v3#382] [Q&A]
自定义工具相册管理选图，返回的数据如何在对应光标位置插入，或者说如何获取当前光标位置？这个editorRef.value?.focus(option)返回都是undefined，怎么获取光标位置，没找到类似案例可以参考[思考]
![image](https://github.com/imzbf/md-editor-v3/assets/57931775/71e438e7-2b2a-48e6-8b56-edd8644578c9)

**Comment by imzbf** [comment]
调用`insert`就可以直接将你需要插入的内容插入到当前光标的位置，不需要你手动去调整位置的

https://imzbf.github.io/md-editor-v3/zh-CN/docs#%F0%9F%92%89%20insert

**Reply by Nabian31** [reply]
调试出来了，被这个参数selectedText误导了，以为一定要从这个参数传进来，但始终是空
![image](https://github.com/imzbf/md-editor-v3/assets/57931775/f198700b-ae48-40c1-8b35-61f240dbe3d3)
后来直接给targetValue赋值，就可以插入内容，呵呵
![image](https://github.com/imzbf/md-editor-v3/assets/57931775/ed436022-3e42-4b7c-a7d9-3020ab495939)
